src/doc/rustdoc/src/lints.md

```diff
@@ -285,3 +285,43 @@ warning: unclosed HTML tag `h1`
 
 warning: 2 warnings emitted
 ```
+
+## automatic_links
+
+This link is **allowed by default** and is **nightly-only**. It detects links
+which could use the "automatic" link syntax. For example:
```

Comment:
Oh hmm - this is also a good improvement but not what I had in mind originally, I see now why you want to call it `automatic_links`. I was imagining something like

```rust
/// http://example.com
pub fn f() {}
```
```
warning: URL which is not a hyperlink
 --> foo.rs:3:5
  |
3 | /// http://example.com
  |     ^^^^^^^^^^^^^^^^^^  this will be text, not clickable, when rendered as HTML
 = help: use an automatic link instead: <http://example.com>
```

How hard do you think it would be to add that? No need to do it here if it's difficult, but I think it would be nice to have.

Reply:
Let me try first. :)